sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -2403,6 +2408,17 @@ pi_result cuda_piKernelCreate(pi_program program, const char *kernel_name,
     retErr = PI_OUT_OF_HOST_MEMORY;
   }
 
+  pi_device device = program->get_context()->get_device();
+  pi_kernel piKernel = retKernel.get();
+  size_t reqdThreadsPerBlock[3] = {};
+  pi_result retError = cuda_piKernelGetGroupInfo(
+      piKernel, device, PI_KERNEL_GROUP_INFO_COMPILE_WORK_GROUP_SIZE,
+      sizeof(reqdThreadsPerBlock), reqdThreadsPerBlock, nullptr);
+  assert(retError == PI_SUCCESS);
+
+  piKernel->save_reqd_threads_per_block(sizeof(reqdThreadsPerBlock),
+                                        reqdThreadsPerBlock);
```

Comment:
Probably it would be more logical if it is done in the constructor of `_pi_kernel` since `reqdThreadsPerBlock_` member is supposed to be always initialized.

Reply:
ok, done.